libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/IDPRequestHandler.java

```diff
@@ -71,9 +71,9 @@ public IDPRequestHandler(SPConfig spConfig, UserAccount userAccount) throws IDPR
         this.spConfig = spConfig;
         this.userAccount = userAccount;
         basicValidation();
-        loginUrl = spConfig.getLoginUrl();
+        loginUrl = userAccount.getLoginServer();
```

Comment:
This one fixes a corner case bug when you swizzle out to the IDP and then decide to change login hosts.